tests/dialects/test_snowflake.py

```diff
@@ -44,6 +44,14 @@ def test_snowflake(self):
         self.validate_all("CAST(x AS CHAR VARYING)", write={"snowflake": "CAST(x AS VARCHAR)"})
         self.validate_all("CAST(x AS CHARACTER VARYING)", write={"snowflake": "CAST(x AS VARCHAR)"})
         self.validate_all("CAST(x AS NCHAR VARYING)", write={"snowflake": "CAST(x AS VARCHAR)"})
+        self.validate_all(
+            "SELECT DATE_PART('year', TIMESTAMP '2020-01-01')",
+            write={
+                "hive": "SELECT EXTRACT(year FROM CAST('2020-01-01' AS TIMESTAMP))",
+                "snowflake": "SELECT DATE_PART('year', CAST('2020-01-01' AS TIMESTAMPNTZ))",
+                "spark": "SELECT EXTRACT(year FROM CAST('2020-01-01' AS TIMESTAMP))",
```

Comment:
Without this fix, `year` would be generated as a string which is invalid Hive / Spark SQL:

```
spark-sql (default)> SELECT EXTRACT('year' FROM CAST('2020-01-01' AS TIMESTAMP));

[PARSE_SYNTAX_ERROR] Syntax error at or near 'FROM'.(line 1, pos 22)

== SQL ==
SELECT EXTRACT('year' FROM CAST('2020-01-01' AS TIMESTAMP))
----------------------^^^
spark-sql (default)> SELECT EXTRACT(year FROM CAST('2020-01-01' AS TIMESTAMP));
2020
Time taken: 4.236 seconds, Fetched 1 row(s)
```